packages/repack/src/modules/ScriptManager/types.ts

```diff
@@ -88,6 +88,25 @@ export interface ScriptLocator {
    * Setting this flat to `false`, disables that behavior.
    */
   cache?: boolean;
+  /**
+   * Function called before loading or getting from the cache and after resolving the script locator.
```

Comment:
It might be a good idea to add one or two sentences here describing that it's the right place in the 'lifecycle' of consuming Script to ask the user if they want to update/download new version of the Script/s